### DISCUSSION TITLE
fix(types): export defineAsyncComponent type

### DISCUSSION THREAD

**Discussion by 202780181** [2392645016/vue#5] [Q&A] [answered]
What kind of change does this PR introduce? (check at least one)

 Bugfix
 Feature
 Code style update
 Refactor
 Build-related changes
 Other, please describe:
Does this PR introduce a breaking change? (check one)

 Yes
 No
If yes, please describe the impact and migration path for existing applications:

The PR fulfills these requirements:

 It's submitted to the dev branch for v2.x (or to a previous version branch), not the master branch
 When resolving a specific issue, it's referenced in the PR's title (e.g. fix #xxx[,#xxx], where "xxx" is the issue number)
 All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
 New/updated tests are included
If adding a new feature, the PR's description includes:

 A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)
Other information:

This PR exports the types for defineAsyncComponent that were previously pending. See https://github.com/vuejs/vue/issues/12608#issuecomment-1180484550.

**Comment by 2392645016** [comment, marked as answer]
@202780181 
Yes
No

1. If yes, please describe the impact and migration path for existing applications: # # #3 

- [ ]
